spec/formatting.md

```diff
@@ -7,10 +7,20 @@ when formatting a message for display in a user interface, or for some later pro
 
 To start, we presume that a _message_ has either been parsed from its syntax
 or created from a data model description.
-If this construction has encountered any _Syntax Errors_ or _Data Model Errors_,
-an appropriate error MUST be emitted and a _fallback value_ MAY be used as the formatting result.
+If the resulting _message_ is not _well-formed_, a _Syntax Error_ is emitted.
+If the resulting _message_ is _well-formed_ but is not _valid_, a _Data Model Error_ is emitted.
 
-Formatting of a _message_ is defined by the following operations:
+The formatting of a _message_ is defined by the following operations,
+starting with _Pattern Selection_:
+
+- **_<dfn>Pattern Selection</dfn>_** determines which of a message's _patterns_ is formatted.
+  For a message with no _selectors_, this is simple as there is only one _pattern_.
+  With _selectors_, this will depend on their resolution.
+
+- **_<dfn>Formatting</dfn>_** takes the resolved values of the selected _pattern_,
```

Comment:
Are _Formatting_ and _Expression and Markup Resolution_ out of order? I think one would need to do the resolution bits before performing the "formatting". The introductory text on line 13 suggests that this is an ordered list of operations, so maybe flip-flop the two?

Reply:
The way I would put it, _Expression and Markup Resolution_ is a dependency for both _Pattern Selection_ and _Formatting_, hence putting them in this order here.

As a separate note, it might be a good idea to move the Pattern Selection examples somewhere else; they're pretty voluminous.

Reply:
Maybe restore line 13 then? More like

> The formatting of a _message_ includes these operations: